packages/devcmd/src/process/ProcessExecutor.ts

```diff
@@ -39,161 +53,231 @@ class SafeConsoleLike implements ConsoleLike {
  */
 export class ProcessExecutor {
   private readonly consoleLike: ConsoleLike;
+  private readonly logger: Logger;
 
   constructor(consoleLike: ConsoleLike) {
     this.consoleLike = new SafeConsoleLike(consoleLike);
+    this.logger = new Logger(this.consoleLike.error.bind(this.consoleLike));
     this.execInTty = this.execInTty.bind(this);
     this.execPiped = this.execPiped.bind(this);
     this.execPipedParallel = this.execPipedParallel.bind(this);
     this.execToString = this.execToString.bind(this);
   }
 
   /**
-   * Executes a process and throws an exception if the exit code is non-zero.
-   * Outputs (stdout/stderr) of the process are sent to our stdout/stderr.
+   * Executes a process and pipes its outputs (stdout/stderr) to the log/error level of
+   * this instance's {@link ConsoleLike}.
    *
-   * @param processInfo Information about the process to execute
-   * @returns A promise that resolves on success and rejects on error
+   * The "nonZeroExitCodeHandling" option of the given {@link ProcessInfo} determines
+   * whether this method throws on a non-zero exit code.
    *
-   * @example
-   * <caption>Running ping 127.0.0.1 on localhost</caption>
+   * **Example:** Running a ping against localhost:
    * ```
-   * try {
-   *   await execPiped({
-   *     command: 'ping',
-   *     args: ['127.0.0.1'],
-   *   });
-   * } catch {}
+   * await execPiped({ command: 'ping', args: ['127.0.0.1'] });
+   * ```
+   *
+   * **Example:** Deleting a Docker container and ignoring non-zero exit codes (e.g. when
+   * container does not exit):
    * ```
+   * await execPiped({
+   *   command: 'docker',
+   *   args: ['rm', '--force', 'myContainerName'],
+   *   options: { nonZeroExitCodeHandling: "printNoticeAndReturn" },
+   * });
+   * ```
+   *
+   * @param processInfo Information about the process to execute
+   * @returns A promise resolving to an object with exit status info once the process exits
    */
-  async execPiped(processInfo: ProcessInfo): Promise<void> {
-    await this.execPipedInternal(processInfo, "");
+  async execPiped(processInfo: ProcessInfo): Promise<NodeExitInfo> {
+    return await this.execPipedInternal(processInfo, "");
```

Comment:
Minor: Do we need to await this promise?

Reply:
no, but it doesn't hurt either :)